keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/normalizers/ApplicationLoadBalancerNormalizer.kt

```diff
@@ -0,0 +1,75 @@
+package com.netflix.spinnaker.keel.ec2.normalizers
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.SPINNAKER_API_V1
+import com.netflix.spinnaker.keel.api.ec2.ApplicationLoadBalancer
+import com.netflix.spinnaker.keel.clouddriver.model.ApplicationLoadBalancerModel
+import com.netflix.spinnaker.keel.plugin.ResourceNormalizer
+import org.springframework.stereotype.Component
+
+@Component
+class ApplicationLoadBalancerNormalizer(
+  private val objectMapper: ObjectMapper
+) : ResourceNormalizer<ApplicationLoadBalancer> {
+  override val apiVersion = SPINNAKER_API_V1.subApi("ec2")
+  override val supportedKind = "alb"
+
+  override fun normalize(resource: Resource<*>): Resource<ApplicationLoadBalancer> {
+    @Suppress("UNCHECKED_CAST")
+    val r = resource as Resource<ApplicationLoadBalancer>
+
+    if (r.spec.targetGroups.any { it.name.length > 32 }) {
+      throw IllegalStateException("targetGroup names have a 32 character limit")
```

Comment:
We definitely need to think about how to model this kind of thing in the schema

Reply:
Looks like precondition checks can be added to a data class' init block, going to test that out.

Reply:
I wasn't suggesting you need to do it as part of this PR, just thinking aloud.